packages/frontend/src/view/components/table/TablePreview.tsx

```diff
@@ -5,49 +5,54 @@ import { Button } from '../Button'
 import { Link } from '../Link'
 import { SectionHeading } from '../SectionHeading'
 
-interface TablePreviewProps {
-  title?: ReactNode
+type TablePreviewProps = {
   path: string
   entryShortNamePlural: string
   entryLongNamePlural: string
   visible: number
   children: ReactNode
-}
+} & (
+  | {
+      title: string
+      viewAllPosition: 'top'
+    }
+  | {
+      viewAllPosition: 'bottom'
+    }
```

Comment:
Not a fan. I'd prefer `title?` and `viewAllPostion: top | bottom`, but I understand the reasoning for this.